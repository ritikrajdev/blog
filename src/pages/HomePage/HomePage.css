.home-page {
  padding-top: 2rem;
  padding-bottom: 2rem;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 2rem;
  justify-content: center;
}
