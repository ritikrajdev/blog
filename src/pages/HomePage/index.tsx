import React, { useContext, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

import Card from '../../components/elements/Card';
import { GET_ALL_BLOGS } from '../../constants/apiEndpoints/blog';

import { makeRequest } from '../../utils/makeRequest';

import './HomePage.css';
import { PostsContext } from '../../contexts/blogContext';

export default function HomePage() {
  const navigate = useNavigate();
  // actually no need of it !!
  const { posts, setPosts } = useContext(PostsContext);

  useEffect(() => {
    makeRequest(GET_ALL_BLOGS, {}, navigate).then((data) => {
      setPosts(data);
    });
  }, []);

  if (posts === undefined)
    return <div className='body-padding home-page'>Loading...</div>;

  return (
    <div className='body-padding home-page' data-testid='cards'>
      {posts.map((post, idx) => (
        <Card key={idx} {...post} />
      ))}
    </div>
  );
}
