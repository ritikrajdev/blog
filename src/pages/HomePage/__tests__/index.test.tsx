import React from 'react';
import { render, waitFor } from '@testing-library/react';

import HomePage from '..';

const mockData = [
  {
    id: 2,
    date: '2023-01-01T07:46:22.001Z',
    // eslint-disable-next-line camelcase
    reading_time: '1 min',
    title: 'How painting outdoors can transform your',
    description:
      'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your readers for more. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc egestas laoreet nibh euismod vestibulum.',
    claps: 1,
    liked: true,
    image: 'https://i.ibb.co/8cDkQkS/outdoor.png',
  },
  {
    id: 3,
    date: '2023-02-20T21:00:00.001Z',
    // eslint-disable-next-line camelcase
    reading_time: '2 mins',
    title: '10 young painters you need to know',
    description:
      'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your readers for more. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc egestas laoreet nibh euismod vestibulum.',
    claps: 15,
    liked: false,
    image: 'https://i.ibb.co/V38cHQ3/young-painters.png',
  },
];

// FIXME: showing axios error if both tests are present
describe('HomePage', () => {
  jest.mock('../../../utils/makeRequest', () => ({
    makeRequest: jest.fn().mockResolvedValue(mockData),
  }));

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should show loading when initially rendered', () => {
    const screen = render(<HomePage />);
    const loadingNode = screen.getByText('Loading...');
    expect(loadingNode).toBeTruthy();
    expect(screen.asFragment()).toMatchSnapshot();
  });

  it('should display 2 cards', () => {
    const screen = render(<HomePage />);
    waitFor(() => {
      const cardContainer = screen.getByTestId('cards');
      expect(cardContainer.childNodes).toHaveLength(2);
    });
  });
});
