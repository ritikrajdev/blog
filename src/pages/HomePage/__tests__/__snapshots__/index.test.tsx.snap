// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should show loading when initially rendered 1`] = `
<DocumentFragment>
  <div
    class="body-padding home-page"
  >
    Loading...
  </div>
</DocumentFragment>
`;
